docs/resources/oxide_instance.md

```diff
@@ -64,41 +71,27 @@ resource "oxide_instance" "example" {
 }
 ```
 
-### Define user data
-
-```hcl
-resource "oxide_instance" "example" {
-  project_id  = "c1dee930-a8e4-11ed-afa1-0242ac120002"
-  description = "a test instance"
-  name        = "myinstance"
-  host_name   = "<host value>"
-  memory      = 1073741824
-  ncpus       = 1
-  user_data   = filebase64("path/to/init.sh")
-}
-```
-
 ## Schema
 
 ### Required
 
 - `description` (String) Description for the instance.
-- `host_name` (String) Host name of the instance.
+- `host_name` (String) Hostname of the instance.
 - `memory` (Number) Instance memory in bytes.
 - `name` (String) Name of the instance.
 - `ncpus` (Number) Number of CPUs allocated for this instance.
 - `project_id` (String) ID of the project that will contain the instance.
 
 ### Optional
 
-- `boot_disk_id` (String, Optional) ID of the disk the instance should be booted from. This ID must also be present in `disk_attachments`.
-- `disk_attachments` (Set of String, Optional) IDs of the disks to be attached to the instance.
-- `external_ips` (Set of Object, Optional) External IP addresses provided to this instance. (see [below for nested schema](#nestedatt--ips))
-- `network_interfaces` (Set of Object, Optional) Virtual network interface devices attached to an instance. (see [below for nested schema](#nestedatt--nics))
-- `ssh_public_keys` (Set of String, Optional) An allowlist of IDs of the saved SSH public keys to be transferred to the instance via cloud-init during instance creation.
-- `start_on_create` (Boolean, Default `true`) Starts the instance on creation when set to true.
-- `timeouts` (Attribute, Optional) (see [below for nested schema](#nestedatt--timeouts))
-- `user_data` (String) User data for instance initialization systems (such as cloud-init). Must be a Base64-encoded string, as specified in [RFC 4648 § 4](https://datatracker.ietf.org/doc/html/rfc4648#section-4) (+ and / characters with padding). Maximum 32 KiB unencoded data.
+- `boot_disk_id` (String, Optional) ID of the disk to boot the instance from. When provided, this ID must also be present in `disk_attachments`.
+- `disk_attachments` (Set of String, Optional) IDs of the disks to be attached to the instance. The first entry in this set will be used to boot the instance unless `boot_disk_id` is provided. 
```

Comment:
The set is an unordered list, so we can't really guarantee that the first entry will be used. The reason we're using a set instead of a list is because the instance disk list endpoint does not return the attached disks in the same order we set them.

Reply:
Whoops! I read it as an ordered list at first. I'll revert. It sparked another question though that I just tested. What happens when multiple disks are specified via `disk_attachments` and only one of them is bootable and it's not the first disk? The answer is that the instance will try to boot from disks in order until it finds a bootable disk or errors.

Given that context should we update this to say something along the lines of when multiple disks are provided use `boot_disk_id` to specify the boot disk?

Reply:
Yeah! That sounds good. And probably warning that otherwise the boot disk will be chosen randomly.